test/Control/Monad/Free/MocksSpec.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Control.Monad.Free.MocksSpec where
+
+import Control.Monad.Free
+import Control.Exception
+import Test.Hspec
+import Prelude hiding (getLine)
+
+import Control.Monad.Free.Mocks
+
+-- * Functor
+data ConsoleF a where
+  GetLine :: (String -> a) -> ConsoleF a
+  WriteLine :: String -> a -> ConsoleF a
+
+deriving instance Functor ConsoleF
+
+instance HasMock ConsoleF where
+  mockExtract (GetLine f) = Right $ W f
+  mockExtract (WriteLine _ a) = Left a
+
+
+instance Show (ConsoleF a) where
+  show (GetLine _) = "GetLine"
+  show (WriteLine _ _) = "WriteLine"
```

Comment:
Ah.sure thing. forgot to change that to ShowConstrctor actually. I was using Show while i was toying around, will change that.